packages/model/addon/-private/model.js

```diff
@@ -1375,34 +1392,41 @@ if (DEBUG) {
     return isBasicDesc(instanceDesc) && lookupDescriptor(obj.constructor, keyName) === null;
   };
 
-  const INSTANCE_DEPRECATIONS = new WeakMap();
-  const DEPRECATED_LIFECYCLE_EVENT_METHODS = [
-    'becameError',
-    'becameInvalid',
-    'didCreate',
-    'didDelete',
-    'didLoad',
-    'didUpdate',
-    'ready',
-    'rolledBack',
-  ];
-
-  let lookupDeprecations = function lookupInstanceDeprecations(instance) {
-    let deprecations = INSTANCE_DEPRECATIONS.get(instance);
-
-    if (!deprecations) {
-      deprecations = new Set();
-      INSTANCE_DEPRECATIONS.set(instance, deprecations);
-    }
+  let lookupDeprecations;
+  let DEPRECATED_LIFECYCLE_EVENT_METHODS;
```

Comment:
Could probably leave this defined as a const and do:

```js
const DEPRECATED_LIFECYLE_EVENT_METHODS = DEPRECATED_RECORD_LIFECYCLE_EVENT_METHODS === true && [
  // ...snip...
];
```

Super minor, feel free to disregard.

Reply:
Also, I find the naming of these two variables to be pretty confusing (one is the array of methods, the other is the deprecation flag) since they are so similarly named. Maybe you can rename the local method name variable?

Reply:
this was an already existing variable but thank you for catching the similarity, I'll rename it.